test_util/helpers.py

```diff
@@ -4,10 +4,34 @@
 import requests
 import retrying
 
+import test_util
+
 Host = namedtuple('Host', ['private_ip', 'public_ip'])
 SshInfo = namedtuple('SshInfo', ['user', 'home_dir'])
 
 
+class DcosUser:
+    """A lightweight user representation."""
+    def __init__(self, auth_json):
+        self.auth_json = auth_json
+        self.auth_header = None
+        self.auth_token = None
+        self.auth_cookie = None
+
+    def authenticate(self, cluster):
+        assert isinstance(cluster, test_util.cluster_api.ClusterApi), 'Unrecognized cluster object!'
+        logging.info('Attempting authentication')
+        r = cluster.post(path='/acs/api/v1/auth/login', disable_suauth=True, payload=self.auth_json)
+        if r.status_code >= 400:
```

Comment:
`r.raise_for_status()` (http://docs.python-requests.org/en/master/user/quickstart/#response-status-codes) I think will give us much better error messages here. Ideally want the status_code,  the message, and the `.content`. The `.content` usually contains lots of juicy info.